File: java-neo4j-driver/social/src/test/java/com/tacitphoenix/neo4j/SocialTest.java
package com.tacitphoenix.neo4j;

import static org.junit.Assert.assertEquals;

import java.util.Arrays;
import java.util.List;

import org.junit.Test;
import org.neo4j.driver.v1.StatementResult;

public class SocialTest {
  @Test
  public void insertFriends() {
    Social social = new Social();
    StatementResult result = social.insertFriends();
    while (result.hasNext()){
        assertEquals(result.next().get("name").asList(), "Anna");
    }
  }

  @Test
  public void friendsOfAFriend() {
    Social social = new Social();
    StatementResult result = social.friendsOfAFriend();
    while (result.hasNext()){
        assertEquals(result.next().get("name").asString(), "Anna");
    }
  }

  @Test
  public void commonFriends() {
    Social social = new Social();
    StatementResult result = social.commonFriends();
    while (result.hasNext()){
        assertEquals(result.next().get("friend").asString(), "Bob");
    }
  }

  @Test
  public void connectingPaths() {
    List names = Arrays.asList("Joe", "Sally", "Anna", "Jim", "Billy");
    Social social = new Social();
    StatementResult result = social.connectingPaths();
    while (result.hasNext()){
        assertEquals(result.next().get("names").asList(), names);
    }
  }
}

File: elixir-neo4j-driver/social/test/social_test.exs
defmodule SocialTest do
  use ExUnit.Case
  doctest Social

  test "insert friends" do
    result = Social.insert_friends()
    assert result.records == []
  end

  test "common friends" do
    result = Social.common_friends()
    assert result.records == [["Bob"]]
  end

  test "friends of a friend" do
    result = Social.friends_of_a_friend()
    assert result.records == [["Anna"]]
  end

  test "connecting paths" do
    result = Social.connecting_paths()
    [[nodes]] = result.records
    assert length(nodes) == 5
  end
end
